Formula/dockly.rb

```diff
@@ -0,0 +1,20 @@
+require "language/node"
+
+class Dockly < Formula
+  desc "Immersive terminal interface for managing docker containers, services and images"
+  homepage "https://github.com/lirantal/dockly#readme"
+  url "https://registry.npmjs.org/dockly/-/dockly-3.21.4.tgz"
+  sha256 "87b2d87ea01eb43fdfd030c775df69249dd2ecae14d4452360d343095cbae943"
+  license "MIT"
+
+  depends_on "node"
+
+  def install
+    system "npm", "install", *Language::Node.std_npm_install_args(libexec)
+    bin.install_symlink Dir["#{libexec}/bin/*"]
+  end
+
+  test do
+    assert_match "dockly #{version} by Liran Tal<liran.tal@gmail.com>", shell_output("#{bin}/dockly --version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
dockly is a GUI program and it's only use is rendering the GUI in the terminal, so I think I can't make any tests rather than checking the version.

@SMillerDev do you have ay other suggestion how I could make a valid test?

Reply:
It looks like a TUI, so you can probably spawn a `pty` and try to interact with the TUI a little from there. Look at other formulae that to `require "pty"` and `PTY.spawn` in the test block for examples.

Reply:
@carlocab TUI is definitely a better way to call it, thanks for the referral! Will look into it now

Reply:
@SMillerDev @carlocab I just pushed a fix now, WDYT?

Reply:
For dockly to display the TUI it would also need a socket interface available of a Docker daemon to connect to. Would it be possible in the test environment to make that available?

Reply:
Firing up a container for the test is a bit heavy, and probably isn't a good idea on Linux, where we're already running inside a container.

Can we point it at a non-existent Docker socket and check that it complains in the right way?

Reply:
I made another fix. We are now testing if dockly is throwing the right error when having an invalid Docker socket.